cluster-autoscaler/cloudprovider/gce/gce_manager_test.go

```diff
@@ -690,7 +690,7 @@ func TestGetMigForInstance(t *testing.T) {
 			buildListInstanceGroupManagersResponsePart(defaultPoolMigName, zoneB, 7),
 			buildListInstanceGroupManagersResponsePart(extraPoolMigName, zoneB, 8),
 		)).Once()
-	server.On("handle", "/projects/project1/zones/us-central1-b/instanceGroupManagers/gke-cluster-1-default-pool/listManagedInstances").Return(buildFourRunningInstancesOnDefaultMigManagedInstancesResponse(zoneB)).Twice()
+	server.On("handle", "/projects/project1/zones/us-central1-b/instanceGroupManagers/gke-cluster-1-default-pool/listManagedInstances").Return(buildFourRunningInstancesOnDefaultMigManagedInstancesResponse(zoneB))
```

Comment:
There should be two API calls here - this test actually detected a bug that this change would introduce. The issue is that in the `fillMigInstances` we check only the `mig to instances` cache, forgetting about the `instances to mig` cache.

The scenario which fails here:
- There is a MIG `A` with instances `a` and `b`
- We already cached all the info about the MIG and its instances
- The instance `c` is added to MIG `A`
- We want to get the MIG for instance `c`
- We don't update the instances of `A` because `a` and `b` are already cached

This should be solved by my other suggestion of moving the lock elsewhere + removing the cache check from `fillMigInstances`.